app/src/main/kotlin/com/simplemobiletools/flashlight/activities/WidgetTorchConfigureActivity.kt

```diff
@@ -3,133 +3,107 @@ package com.simplemobiletools.flashlight.activities
-        initVariables()
 
         val isCustomizingColors = intent.extras?.getBoolean(IS_CUSTOMIZING_COLORS) ?: false
-        mWidgetId = intent.extras?.getInt(AppWidgetManager.EXTRA_APPWIDGET_ID) ?: AppWidgetManager.INVALID_APPWIDGET_ID
+        viewModel.setWidgetId(intent.extras?.getInt(AppWidgetManager.EXTRA_APPWIDGET_ID) ?: AppWidgetManager.INVALID_APPWIDGET_ID)
 
-        if (mWidgetId == AppWidgetManager.INVALID_APPWIDGET_ID && !isCustomizingColors) {
+        if (viewModel.widgetId.value == AppWidgetManager.INVALID_APPWIDGET_ID && !isCustomizingColors) {
             finish()
         }
 
-        binding.apply {
-            configSave.setOnClickListener { saveConfig() }
-            configWidgetColor.setOnClickListener { pickBackgroundColor() }
-
-            val primaryColor = getProperPrimaryColor()
-            configWidgetSeekbar.setColors(getProperTextColor(), primaryColor, primaryColor)
+        enableEdgeToEdgeSimple()
+        setContent {
+            AppThemeSurface {
+                val widgetColor by viewModel.widgetColor.collectAsStateWithLifecycle()
+                val widgetAlpha by viewModel.widgetAlpha.collectAsStateWithLifecycle()
 
-            if (!isCustomizingColors && !isOrWasThankYouInstalled()) {
-                mFeatureLockedDialog = FeatureLockedDialog(this@WidgetTorchConfigureActivity) {
-                    if (!isOrWasThankYouInstalled()) {
-                        finish()
-                    }
+                val colorPickerDialogState = rememberAlertDialogState().apply {
+                    ColorPicker(this)
                 }
+
+                WidgetConfigureScreen(
+                    widgetDrawable = R.drawable.ic_flashlight_vector,
+                    widgetColor = widgetColor,
+                    widgetAlpha = widgetAlpha,
+                    onSliderChanged = viewModel::changeAlpha,
+                    onColorPressed = colorPickerDialogState::show,
+                    onSavePressed = ::saveConfig
+                )
+
+                CheckFeatureLocked(skipCheck = isCustomizingColors)
             }
+        }
+    }
 
-            configSave.backgroundTintList = ColorStateList.valueOf(getProperPrimaryColor())
-            configSave.setTextColor(getProperPrimaryColor().getContrastColor())
+    @Composable
+    private fun ColorPicker(
+        alertDialogState: AlertDialogState
+    ) {
+        val brightDisplayColor by viewModel.widgetColor.collectAsStateWithLifecycle()
+        alertDialogState.DialogMember {
+            ColorPickerAlertDialog(
+                alertDialogState = alertDialogState,
+                color = brightDisplayColor,
+                removeDimmedBackground = true,
+                onActiveColorChange = {},
+                onButtonPressed = { wasPositivePressed, color ->
+                    if (wasPositivePressed) {
+                        viewModel.updateColor(color)
+                    }
+                }
+            )
```

Comment:
you can do something like 

```kotlin
@Composable
    private fun getColorPickerDialogState() =
        rememberAlertDialogState().apply {
            DialogMember {
                val brightDisplayColor by viewModel.widgetColor.collectAsStateWithLifecycle()
                ColorPickerAlertDialog(
                    alertDialogState = this,
                    color = brightDisplayColor,
                    removeDimmedBackground = true,
                    onActiveColorChange = {},
                    onButtonPressed = { wasPositivePressed, color ->
                        if (wasPositivePressed) {
                            viewModel.updateColor(color)
                        }
                    }
                )
            }
        }
```

Which is more readable and have

```kotlin
val colorPickerDialogState = getColorPickerDialogState()
```

Reply:
I messed it up while I was trying to sort out recomposition issues in `BrightDisplayActivity` and forgot to clean it up after I was done :smile: